js/containers/dashboard/dashboard-app.js

```diff
@@ -32,12 +33,20 @@ class DashboardApp extends PureComponent {
 
   render () {
     const { initialLoading } = this.state
-    const { error, reportTree, students, lastUpdated, activityProgress } = this.props
+    const { error, reportTree, students, lastUpdated, studentProgress, expandedStudents, expandedActivities, setActivityExpanded, setStudentExpanded } = this.props
```

Comment:
I'm not sure why this is confusing. It's pretty common to pass handler / functions as props. Also, `mapDispatchToProps` is one of the standard Redux functions. Redux is a bit overspoken. :wink:
I'd say there's **almost** no difference here and matter of taste, but actually there's one significant difference:

`setActivityExpanded={(activityId, value) => dispatch(setActivityExpanded(activityId, value))}` craetes a new function each time you render a component. It means that the child component will **always** think that some property has changed and trigger the whole rendering machinery. This is obviously bad. It can have some bad pefrormance consequences if you're unlucky (if you follow this pattern everywhere, you can re-render the whole react tree all the time). It's not even related to `dispatch` itself, but passing `() => { }` in React props.

Reply:
Thanks for the explanation Piotr! Like I said, this is my first time really working with Redux so everything is a little confusing :) I think the main thing that tripped me up was the fact that `setActivityExpanded` is used as the name for two different functions in the same file: wrapped by dispatch in one place, but not in another. Giving the functions different names might have helped - but, it seems like your names are standard practice, and I'm sure I'll get used to it.

I took the idea from the intro Redux tutorial...they only have one example of a combo component/container (https://github.com/reduxjs/redux/blob/master/examples/todos/src/containers/AddTodo.js). But I see the difference - here we're passing functions to a prop, which would definitely kill rendering! Of course, the function could be created once outside of the `render` function...but that's essentially what `mapDispatchToProps` is doing anyway ;)

Reply:
Personally, I don't like what they're doing in this example. 😉 
I usually create a regular method and bind it in constructor like this:
```javascript
constructor (props) {
  super(props)
  this.onSubmit = this.onSubmit.bind(this)
}

onSubmit () { ... }

render () {
  return (
    <form onSubmit={this.onSubmit}
    (...)
```

It is not important in case or regular DOM elements like `<form>`, as obviously there won't be anything down in the component tree. But if it was a React component that had large tree of subcomponents and some of them were doing expensive calculations in lifecycle methods, it can cause unnecessary performance problems. In 99% cases it'll be fine, but I prefer to keep it like this, just not to worry about this 1%.

Reply:
If you define `onSubmit` as an arrow function, you can avoid the explicit bind in the constructor:
```javascript
onSubmit = () => { ... }
```